src/core_plugins/metric_vis/public/metric_vis_controller.js

```diff
@@ -54,6 +56,14 @@ export class MetricVisComponent extends Component {
     return colors[label];
   }
 
+  _needsLightText(bgColor) {
+    const color = /rgb\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\)/.exec(bgColor);
```

Comment:
do we really need this? How does it arise that we don't know bgColor if of `rgb()` format?

Reply:
The colors are currently calculated by the [`getHeatmapColors`](https://github.com/elastic/kibana/blob/master/src/ui/public/vislib/components/color/heatmap_color.js#L53) function, which will always return a color in "rgb(...)" format to us.

This part will hopefully be cleaner once we build a proper palette color picker later on.

Reply:
Okay I guess I misread. We DO know that the color is in `rgb()` format. That's why we parse it here, into it's red, green and blue component, so we can pass three numeric values to the `isDarkColor` method.